data/loaders.py

```diff
@@ -104,6 +115,51 @@ def load(self, verbose: bool = True):
         if verbose:
             print('Done.')
 
+
+    def load_v2(self, verbose: bool = True, subset_size: int = None):
+        train_set_size_ratio = 0.7
+
+        transformation = transforms.Compose([
+            transforms.Resize(self.resize_to),
+            transforms.CenterCrop(224),
+            transforms.ToTensor(),
+            transforms.Normalize(mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5])
+        ])
+        
+        if verbose:
+            print('Loading dataset from files...')
+        full_dataset = torchvision.datasets.ImageFolder(
+            root=self.images_dir,
+            transform=transformation
+        )
+        
+        self.classes_count = len(full_dataset.classes)
+        self.label_names = dict([(value, key) for key, value in full_dataset.class_to_idx.items()])
+
+        if subset_size:
+            full_dataset = torch.utils.data.Subset(full_dataset, np.random.choice(len(full_dataset), subset_size, replace=False))
+    
+
+        train_size = int(train_set_size_ratio * len(full_dataset))
+        test_size = len(full_dataset) - train_size
+        train_dataset, test_dataset = torch.utils.data.random_split(full_dataset, [train_size, test_size])
```

Comment:
Nie wiem czy to jest najlepszy pomysł zeby te zbiory losować. Zależnie od tego jaki mamy seed możemy dostać różne rezultaty, wg mnie powinniśmy po prostu pierwsze x wartości brać jako trening a reszta jako test.

Reply:
Ja osobiście zostałbym przy tej metodzie, przewija się w  większości tutoriali przy ładowniu danych. Zgadzam się odnośnie seedu, w obecnej implementacji posplitowany dataset będzie niereprodukowalny. Do metody przekazałem obiekt generator z manual_seed(0).